source/administrator-guide/managing-users-and-groups/authentication-mode.rst

```diff
@@ -510,6 +511,291 @@ You can configure your environment by updating the previous file or by defining
+
+Setup Application Roles;
+
+#. In Application you created, go to "App Roles".
+#. Add new Groups - "Editor", "Reviewer", etc...
+
+Assign Users:
+
+#. Go to Azure AD, Enterprise Application, then the application you created
+#. Choose "Assign users and groups"
+#. Press the "Add user/group" (top)
+#. Press "None Selected" (under Users) and choose some users
+#. Press "None Selected" (Under Select a Role) and choose some roles
+#. Configure all your users with roles
+
+
+Your environment variables will looks like this;
+
+.. code-block:: properties
+
+    GEONETWORK_SECURITY_TYPE=openidconnect
+    OPENIDCONNECT_CLIENTSECRET='...'
+    OPENIDCONNECT_CLIENTID='...'   
+    OPENIDCONNECT_SERVERMETADATA_JSON_TEXT='...big json text...' 
+    OPENIDCONNECT_IDTOKENROLELOCATION='roles'
+
+NOTE: The roles are in the "roles" part of the ID Token.
+
+
+NOTE: You don't typically have to do any role conversion since the role name will be used in the ID Token.
+
+OIDC Bearer Tokens
+==================
+
+Bearer Tokens are also supported - you can attach the JWT Bearer token to any request by setting the HTTP header like this;
+
+.. code-block:: properties
+    
+     Authorization: Bearer:  <JWT token>
+
+Bearer Tokens are mostly used for automated (desktop or application) API calls - real users should just login normally using OIDC.
+
+#. Setup your OIDC configuration (see above)
+#. Setup the OIDC Bearer token configuration (see below)
+#. Obtain a Bearer token from the OIDC server.  This is the hard part and there are several ways to do this.  One way that is used is via the OAuth 2.0 Device Authorization Grant ("Device Flow") workflow.
+#. Attach it to your request headers (see above)
+#. Make protected requests to the Geonetwork API
+
+This has been tested with Keycloak and with Azure AD.  It should work with other JWT-based OIDC services.
+
+Validation   
+``````````
+
+The token is validated in three major ways;
+
+#. The bearer token will be used to access the `userinfo` ("token validation") endpoint specified in the OIDC configuration.  This means the IDP validates the token (at the very least its signature and expiry).
+#. The bearer token (JWT) will be checked that the audience for it is the same as our configurated OIDC configuration.  This will ensure that someone isn't getting a token from a different service and attempting to use it here.  See `AudienceAccessTokenValidator.java`
+#. The bearer token (JWT) will be checked that the subject of the JWT and the `userinfo` (returned from the IDP) are the same.  This shouldnt be a problem in our use-case, but the OAUTH2 specification recommends this check.  See `SubjectAccessTokenValidator.java`
+
+
+Configuration 
+`````````````
+
+Configure OIDC as above - ensure this is working.
+
+Instead of using `GEONETWORK_SECURITY_TYPE=openidconnect`, use `GEONETWORK_SECURITY_TYPE=openidconnectbearer`.
+
+Inside `config-security-openidconnectbearer.xml`;
```

Comment:
Better to use to format the file names `` instead of `.

Reference them inside the `WEB-INF/config-security` folder as in other places in the document

Reply:
changed